v1.c

```diff
@@ -0,0 +1,44 @@
+//irq.h
+int32_t irq_register(struct irq_ctrl_desc *desc, uint32_t irq_id,
```

Comment:
what is irq_id ? is it a unique interrupt identifier number from the interrupt controller table ?

Reply:
yes